internal/cmd/dopplertemplatefuncs.go

```diff
@@ -0,0 +1,132 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"unicode"
+
+	"github.com/coreos/go-semver/semver"
+	"golang.org/x/exp/slices"
+
+	"github.com/twpayne/chezmoi/v2/internal/chezmoilog"
+)
+
+type dopplerConfig struct {
+	Command     string   `json:"command" mapstructure:"command" yaml:"command"`
+	Args        []string `json:"args"    mapstructure:"args"    yaml:"args"`
+	Project     string   `json:"project" mapstructure:"project" yaml:"project"`
+	Config      string   `json:"config"  mapstructure:"config"  yaml:"config"`
+	outputCache map[string][]byte
+}
+
+var dopplerMinVersion = semver.Version{
+	Major: 0,
+	Minor: 0,
+	Patch: 0,
+} // no specific version required (as of yet)
+
+func toUpperSnakeCase(s string) string {
```

Comment:
This feels too magic to me and should either (a) not do any transformations (can any doppler secret names be lowercase? if not, I think that’s a bug on their side) or (b) only do uppercase transformations. We should not treat `"foo bar"`, `"foo!bar"`, and `"foo_bar"` the same, and since `foo_bar` is a valid shell variable name, we probably shouldn’t treat it the same as `FOO_BAR`.

Reply:
I went backwards and forwards on this one a bit. Doppler, itself, does not support any project secrets not in `UPPER_SNAKE_CASE`. I believe its because they have different formats they want to adhere to (most notably `.env`)

```console
> doppler secrets get --project chezmoi --config dev --plain "not upper snake case"
Unable to fetch secrets
Doppler Error: Secret name "not upper snake case" may only contain uppercase letters, numbers, and underscores.
```

I built this in as a convenience. I can remove if its going to cause too much confusion?

Reply:
I think that it will be better to remove it, especially with the `dopplerProjectJson` function returning the JSON of all secrets in a project and therefore only supporting `UPPER_SNAKE_CASE`.